### PR TITLE
[windows] install ssh through msi

### PR DESCRIPTION
**Pull request by pducolin**
What does this PR do?
---------------------

Install SSH on Windows Azure and AWS instances through the MSI

Which scenarios this will impact?
-------------------

Windows

Motivation
----------

Ensure compatibility across Windows versions, as `Add-WindowsCapability` installs different version depending on the windows version (Windows Server 2019 vs 2022)

Additional Notes
----------------

Suggested by @clarkb7 

Tested with:

- Windows Server 2019 ECS
- Windows latest
- Azure latest

### CODE REVIEW

File: components/os/scripts/setup-ssh.ps1 [hunk 1]
```diff
@@ -3,22 +3,34 @@ $service = Get-Service -Name sshd -ErrorAction SilentlyContinue
...
-if (-not (Get-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0  | Where-Object { $_.State -eq 'Installed' })) {
-  Write-Host "Add OpenSSH Server capability"
-  Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
 } else {
-  Write-Host "OpenSSH Server capability already installed"
+  Write-Host "sshd service not found, installing OpenSSH Server"
+    # Add-WindowsCapability does NOT install a consistent version across Windows versions, this lead to
+    # compatability issues (different command line quoting rules).
+    # Prefer installing sshd via MSI  
+    start-process -passthru -wait msiexec.exe -args '/i https://github.com/PowerShell/Win32-OpenSSH/releases/download/v9.5.0.0p1-Beta/OpenSSH-Win64-v9.5.0.0.msi /qn'
```
Comment (clarkb7) [inline components/os/scripts/setup-ssh.ps1:11]: do you want to fail early if SSH fails to install? if so check the exit code.

```suggestion
    $p=start-process -passthru -wait msiexec.exe -args '/i https://github.com/PowerShell/Win32-OpenSSH/releases/download/v9.5.0.0p1-Beta/OpenSSH-Win64-v9.5.0.0.msi /qn'
    if ($p.ExitCode -ne 0) {
        throw "SSH install failed: $($p.ExitCode)"
    }
```

**Review by clarkb7 — APPROVED** [review]
thanks for doing this!

note: since the last command is `Write-Output` i'm not sure if this script will ever return an error/failure to pulumi. to ensure this, you could make the last command `Start-Service`, or save its result.

**Review by FlorentClarret — APPROVED** [review]

**Comment by pducolin** [comment]
@clarkb7 thank you for the suggestion, I added error handling to all the calls in the script, it would help debugging failing setups. Testing it again on the 3 AMIs I used for the first test

**Comment by pducolin** [comment]
Tested against 
* aws custom AMI with ssh already installed
* aws latest windows
* aws Windows Server 2019
* azure latest windows

**Pull request merged**